src/v2/conversions.rs

```diff
@@ -914,19 +947,51 @@ impl TryFrom<AccountInfo> for super::types::AccountInfo {
             index,
             stake,
             address,
+            cooldowns,
+            available_balance,
         } = value;
         let account_nonce = sequence_number.require()?.into();
         let account_amount = amount.require()?.into();
-        let account_release_schedule = schedule.require()?.try_into()?;
+        let account_release_schedule: AccountReleaseSchedule = schedule.require()?.try_into()?;
         let account_threshold = threshold.require()?.try_into()?;
         let account_encrypted_amount = encrypted_balance.require()?.try_into()?;
         let account_encryption_key = encryption_key.require()?.try_into()?;
         let account_index = index.require()?.into();
-        let account_stake = match stake {
+        let account_stake: Option<super::types::AccountStakingInfo> = match stake {
             Some(stake) => Some(stake.try_into()?),
             None => None,
         };
         let account_address = address.require()?.try_into()?;
+        let mut cds: Vec<super::types::Cooldown> = Vec::with_capacity(cooldowns.len());
+        for cooldown in cooldowns.into_iter() {
+            cds.push(cooldown.try_into()?)
+        }
+        let cooldowns = cds;
+
+        // The available balance is only provided as convenience and in case the
+        // calculation of it changes in the future. It should be present if the node
+        // uses protocol version 7 (or later). If the available balance is not present,
+        // we calculate it manually instead.
```

Comment:
The comment seems to be off a bit, it will be present when the node software is v7 or later, independent of the protocol version.

It would also be good to add, that this fallback calculation can be removed if we ever set the minimum node compatibility version to v7 or higher.

Reply:
>it will be present when the node software is v7 or later, independent of the protocol version.

You mean if a node is running node version 7 but using an older protocol? Is that possible? Or am I misunderstanding. I'm not sure I really understand the difference between what I wrote and what you say here, i.e. the difference between "a node using protocol version 7 (or later)" and "a node [version] 7 or later".

Perhaps you could suggest a wording?

Reply:
```suggestion
        // The available balance is only provided as convenience and in case the
        // calculation of it changes in the future. It should be present if the node
        // is version 7 (or later). If the available balance is not present,
        // we calculate it manually instead.
```


> You mean if a node is running node version 7 but using an older protocol? Is that possible?

Yes, this will happen. So node runners should update their nodes to software version 7 while the chain is running on protocol version 6.
Then a special transaction signed by the governance committee will signal to the chain when exactly the nodes should switch from protocol version 6 to 7.

Reply:
Ah I see, cool!